compiler/rustc_typeck/src/check/wfcheck.rs

```diff
@@ -1087,14 +1087,13 @@ fn check_method_receiver<'fcx, 'tcx>(
     debug!("check_method_receiver: sig={:?}", sig);
 
     let self_ty = fcx.normalize_associated_types_in(span, self_ty);
-    let self_ty =
-        fcx.tcx.liberate_late_bound_regions(method.def_id, ty::Binder::bind(self_ty, fcx.tcx));
+    let self_ty = fcx.tcx.liberate_late_bound_regions(method.def_id, ty::Binder::dummy(self_ty));
 
     let receiver_ty = sig.inputs()[0];
 
     let receiver_ty = fcx.normalize_associated_types_in(span, receiver_ty);
     let receiver_ty =
-        fcx.tcx.liberate_late_bound_regions(method.def_id, ty::Binder::bind(receiver_ty, fcx.tcx));
+        fcx.tcx.liberate_late_bound_regions(method.def_id, ty::Binder::dummy(receiver_ty));
```

Comment:
These again are a bit sus: There aren't late-bound vars, but we're calling `liberate_late_bound_regions`? (mental note to future me: this bit of code is touched in #85499, did we remove these calls?)

Reply:
Okay yeah, we `liberate_late_bound_regions` above, we shouldn't need to do that again here.